python/cudf/cudf/core/multiindex.py

```diff
@@ -859,7 +859,9 @@ def _get_row_major(
             list(row_tuple) if isinstance(row_tuple, tuple) else row_tuple
         ):
             return df[row_tuple]
-        if isinstance(row_tuple, slice):
+        if isinstance(row_tuple, str):
```

Comment:
Hmm wouldn't this be an issue generally if a scalar is passed? e.g.

```
df = pd.DataFrame([[1],[2]], index=[[1, 1],[2,3]])
df.loc[1]
```

Reply:
That already works since take below can handle integer like inputs(`1`)

Reply:
If it understand `_get_valid_indices_by_tuple` and then `take`, these will treat integers as positional when in `loc` we want to treat integers as labels?

Reply:
Yeah, that's what is happening incase of MultiIndex for both pandas & cudf(matching pandas)

Reply:
On pandas 1.5 integers are treated as labels when using `loc` with a MultiIndex

```
>>> import pandas as pd
pd.__version__>>> pd.__version__
'1.5.2'
>>> df = pd.DataFrame([[1],[2]], index=[[-1, -2], [2, 3]])
>>> df
      0
-1 2  1
-2 3  2
>>> df.loc[-1]
   0
2  1
>>> df.loc[1]
KeyError: 1
```

Reply:
Thanks @mroeschke, I've discovered that the above example doesn't work correctly in `cudf` and fixed it in this PR properly. Added parameters for the same test coverage. This should be ready for review.